pkg/payload/task.go

```diff
@@ -119,6 +119,9 @@ type UpdateError struct {
 }
 
 func (e *UpdateError) Error() string {
+	if e.Nested != nil {
```

Comment:
consumers must call cause to find internal errors right?

Reply:
> consumers must call cause to find internal errors right?

That's how master works.  This PR is exposing the cause string as part of the the usual `Error()` rendering, because consumers like our logging and Event creation code are currently not bothering to look at `Cause()`.  Should I be updating them to look at `Cause()` instead?  I can't think of a case where someone would call `Error()` and _not_ be interested in the root cause.

Reply:
An alternative approach would be to stick to `Message` for `Error()` but to add a new `Format()` so folks could use `%s`, `%+v`, etc. to easily switch between different renderings.  For an example implementation, see [`pkg/errors`][1].

[1]: https://github.com/pkg/errors/blob/v0.8.1/errors.go#L127-L141